# Bioinformatics tutorials in fungal genomics

This series of tutorials was designed for undergraduate students with no prior bioinformatics experience. They use the Google Cloud platform, which increases accessiblity, regardless of locally available compute resources, and provides learning opportunities in cloud computing and virtual machins. Fungal genomic data are used throughout the tutorials because fungi have small genomes relative to other Eukaryotes thus reducing files sizes and compute time, yet they provide a higher level complexity than bacterial genomic data. Furthermore, fungal data provide interesting avenues for extended student project work with a foundation in ecology and evolution as they engage in complex symbioses, have genomes with a wide diversity of biosynthetic gene clusters, and often host complex microbiomes. 

The tutorials for which this course was created was offered at a regional comprehensive primarily undergraduate institition, Eastern Washington University. Beyond these tutorials the course incorporated paper presentations and discussions, lectures, and a final group project. 

By the end of the course students will be able to: 
* Comfortably navigate with the basics of working in a Linux-based command line environment 
* Run programs in command environments, including setting options, and defining inputs and outputs
* Carefully examine and evaluate input and output files and data, with an emphasis on common sequence data file types
* Implement thorough documentation practices 
* Understand the basics of blast, multiple-sequence assembly, phylogenetics, genome sequence assembly, and genome sequence annotation
* Communicate bioinformatics-based research verbally and in writing
* Gain independence to run analyses and use tools not specifically covered during the course
* Build and run analyses on virtual machines in a cloud computing environment
* Use conda to install and manage packages

### Acknowledgements

Creation of these tutorials was supported by The National Science Foundation DEB #2115191 and Eastern Washington University. Any opinions, findings, and conclusions or recommendations expressed in this material are those of the author(s) and do not necessarily reflect the views of the National Science Foundation.

<img src="https://user-images.githubusercontent.com/17323363/179559477-c3fdfa60-9886-405b-9aea-124c9305ad88.png" width="100"> <img src="https://user-images.githubusercontent.com/17323363/179559504-9ec579ac-a34e-4279-8472-f3f17df77025.png" width="300">

